src/riak_kv_wm_stats.erl

```diff
@@ -66,25 +69,67 @@ content_types_provided(ReqData, Context) ->
       {"text/plain", pretty_print}],
      ReqData, Context}.
 
-
 service_available(ReqData, Ctx) ->
     {true, ReqData, Ctx}.
 
+malformed_request(RD, Ctx) ->
+    case wrq:get_qs_value("timeout", RD) of
+        undefined ->
+            {false, RD, Ctx};
+        TimeoutStr ->
+            try
+                case list_to_integer(TimeoutStr) of
+                    Timeout when Timeout > 0 ->
+                        {false, RD, Ctx#ctx{timeout=Timeout}}
+                end
+            catch
+                _:_ ->
+                    {true,
+                        wrq:append_to_resp_body(
+                            io_lib:format(
+                                "Bad timeout value ~0p",
```

Comment:
Add in explanation "should be provided in miliseconds"
There is an argument for letting `0` be a valid value for timeout, basically stating, "asap".

Reply:
Added statement of what was expected (milliseconds > 0).  I don't want to support 0, as different people may expect it to mean different things (I would assume never timeout)